tests/PySys/mqtt_port/mqtt_port_change_connection_works/run.py

```diff
@@ -0,0 +1,132 @@
+import sys
+
+sys.path.append("environments")
+from environment_roundtrip_c8y import Environment_roundtrip_c8y
+
+from pysys.basetest import BaseTest
+
+import time
+
+"""
+Validate changing the mqtt port using the tedge command
+
+Given a configured system, that is configured with certificate created and registered in a cloud
+When the thin edge device is disconnected from cloud, sudo tedge disconnect c8y/az
+When mqtt.port is set using tedge with sudo
+When the thin edge device is connected to cloud, sudo tedge connect c8y/az
+Now validate the services that use the mqtt port
+   Validate tedge config set mqtt.port
+   Validate tedge mqtt pub/sub
+   Validate tedge connect c8y/az --test
+   Validate tedge_mapper status
+
+"""
+
+class MqttPortChangeConnectionWorks(Environment_roundtrip_c8y):
+    def setup(self):
+        self.tedge = "/usr/bin/tedge"
+        self.sudo = "/usr/bin/sudo"
+
+        # disconnect from c8y cloud
+        disconnect_c8y = self.startProcess(
+            command=self.sudo,
+            arguments=[self.tedge, "disconnect", "c8y"],
+            stdouterr="disconnect_c8y",
+        )
+        self.addCleanupFunction(self.mqtt_cleanup)
+
+    def execute(self):
+        # set a new mqtt port for local communication
+        mqtt_port = self.startProcess(
+            command=self.sudo,
+            arguments=[self.tedge, "config", "set", "mqtt.port", "8889"],
+            stdouterr="mqtt_port",
+        )
+
+        # wait for a while
+        time.sleep(0.1)
+
+        # connect to c8y cloud
+        connect_c8y = self.startProcess(
+            command=self.sudo,
+            arguments=[self.tedge, "connect", "c8y"],
+            stdouterr="connect_c8y",
+        )
+
+    def validate(self):
+        # validate the mqtt port set
+        self.validate_mqtt_port_set()
+        # validate tedge mqtt pub/sub
+        self.validate_tedge_mqtt()
+        # validate c8y connection
+        self.assertGrep("connect_c8y.out",
+                        "connection check is successful", contains=True)
+        # validate c8y mapper
+        self.validate_tedge_mapper_c8y()
+
+    def validate_mqtt_port_set(self):
+        # subscribe for messages
+        tedge_get = self.startProcess(
+            command=self.sudo,
+            arguments=[self.tedge, "config", "list"],
+            stdouterr="tedge_get",
+        )
+
+        self.assertGrep(
+            "tedge_get.out", "8889", contains=True)
```

Comment:
Better check here for the complete string "mqtt.port=8889"

Reply:
done